hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobUtils.java

```diff
@@ -314,20 +316,24 @@ public static void cleanExpiredMobFiles(FileSystem fs, Configuration conf, Table
           }
           filesToClean
             .add(new HStoreFile(fs, file.getPath(), conf, cacheConfig, BloomType.NONE, true));
+          if (
+            filesToClean.size() > conf.getInt(MOB_CLEANER_BATCH_SIZE_UPPER_BOUND,
+              DEFAULT_MOB_CLEANER_BATCH_SIZE_UPPER_BOUND)
+          ) {
+            try {
+              removeMobFiles(conf, fs, tableName, mobTableDir, columnDescriptor.getName(),
+                filesToClean);
+              LOG.info("Table {} {} expired mob files are deleted", tableName, filesToClean.size());
+            } catch (IOException e) {
+              LOG.error("Failed to delete the mob files, table {}", tableName, e);
+            }
+            filesToClean.clear();
+          }
         }
       } catch (Exception e) {
         LOG.error("Cannot parse the fileName " + fileName, e);
       }
     }
-    if (!filesToClean.isEmpty()) {
-      try {
-        removeMobFiles(conf, fs, tableName, mobTableDir, columnDescriptor.getName(), filesToClean);
-        deletedFileCount = filesToClean.size();
-      } catch (IOException e) {
-        LOG.error("Failed to delete the mob files " + filesToClean, e);
-      }
-    }
-    LOG.info("{} expired mob files are deleted", deletedFileCount);
```

Comment:
Let's still count the deleted files count and output it in the last? We can also output everytime when we delete a batch.